lib/fluent/logger/fluent_logger.rb

```diff
@@ -272,7 +276,19 @@ def send_data(data)
         #  end
         #  data = data[n..-1]
         #end
-        true
+      end
+
+      def send_data_nonblock(data)
+        written = @con.write_nonblock(data)
+        remaining = data.bytesize - written
+
+        while remaining > 0
+          len = @con.write_nonblock(data.byteslice(written, remaining))
```

Comment:
should check sliced data is valid msgpack or not?

Reply:
You mean that the value `data.byteslice(written, remaining)` returns should be checked if it is valid msgpack or not?  Are there any problems if this value is not valid msgpack?

Reply:
Ah, I forgot this write uses same socket.